src/main/java/seedu/contax/model/tag/UniqueTagList.java

```diff
@@ -69,6 +69,19 @@ public void setTags(List<Tag> tags) {
         internalList.setAll(tags);
     }
 
+    public void setTag(Tag target, Tag editedTag) {
+        requireNonNull(target);
+        requireNonNull(editedTag);
+
+        // Checks if edited Tag exists
+        if (contains(editedTag)) {
+            throw new DuplicateTagException();
+        }
```

Comment:
This will fail if `target == editedTag`. You need to ignore the `target` tag.

Reply:
Fixed in 7ff3cf68